### ISSUE TITLE
merge and launch

### ISSUE THREAD

**Issue by csimmons44** [OPLBPO/opl_storybook#17]
The tentative launch date is next Wednesday, which I've asked to push back one week. So assuming we're down the final small bits this week, can we look at merging next week?

**Comment by minorOffense** [comment]
I'm hoping that frank and Wendy can merge the changes today and test deploy on our server then qa by end of week. 

Then you can deploy it to prod whenever you're ready.

**Comment by wendywarren** [comment]
Devin sent me the view for the events which I'm looking into now. He'll be sending along the content type for recipes this afternoon.

So with that and adding the image to accompany each book taxonomy term and a few other small changes, I would say we will be merging everything tomorrow morning. 

I think that still works with Mat's plan of having everything tested on our server and ready for you by end of week.

**Comment by crawleyde** [comment]
Is there a chance we can have QA done by Thursday? Our staff are anxious to launch by the end of this week. We'd prefer to launch in the afternoon on Thursday, if that's possible.

**Comment by minorOffense** [comment]
I'll check with Wendy but I think she was pretty close with the merge. I'll update you this aft.

**Comment by crawleyde** [comment]
OK, thanks.

**Comment by wendywarren** [comment]
@crawleyde  I am just finishing up the top nav, update the make file and merge the updates to the theme this afternoon with all of the changes. 

I will let you know when you can use the make file to build everything this afternoon. I hope that works.

**Comment by crawleyde** [comment]
That's great, thanks a lot.

**Comment by wendywarren** [comment]
@crawleyde I've merged my oplstorybook branch with 7.x for oplbootstrap, and updated the opl_storybook custom module with the latest. I'm going to try the build out on my end to make sure you'll have everything. Wanted to give you an update. :)

**Comment by wendywarren** [comment]
Morning @crawleyde! Everything should be good and ready for you to make a build on your dev server. I've included storybook deploy instructions below to follow once the build is complete. 

- [x] Enable the opl_storybook module - it will enable opl_storybook, videojs, video_embed_field, mediaelement, fitvids.
- [x] Enable image_field_caption (I will add this to the module later)
- [x] Go to admin/structure/pages and enable the Storybook page. 
- [x] You should see three variants already, you can delete the last two variants and remove the content for Food Literacy - Comfort Food … I must have exported with the content saved.
- [x] Go to admin/structure/pages/entity_panels and enable view mode Storybook for Audio, Instagram Feed, Recipe, Storybook Article, Storybook Collections, Storybook Single Image, Storybook Video Intro and Video.
- [x] Go to admin/structure/taxonomy/storybook and add Storybook Terms for Book and Chapter (including a la carte logo for Food Literacy (svg attached) and thumbnail (must be 142 x 80px) for Comfort Food.
- [x] Back to Storybook Page  admin/structure/pages/edit/page-storybook, confirm Selection rules for Comfort Food book and chapter are ‘Food Literacy’ and Comfort Food’.
- [x] Go to /storybook/Food%20Literacy/Comfort%20Food and you should see the top bar with OPL logo, storybook menu with a la carte logo beside Food Literacy, link to French, return to OPL and the Ontario logo.

Let me know if you run into any issues at all. I'll attach the pdf guide. Update to the translate and panel styles part is not included but can get that to you today too.

**Comment by wendywarren** [comment]
Apparently I can't send svg over here. Email coming with logo. Here is the pdf. :)

[oplstorybook-012417.pdf](https://github.com/OPLBPO/opl_storybook/files/742337/oplstorybook-012417.pdf)

FYI ... When you update the pane styles for each piece of content using Storybook options (width, border, etc..) you will notice you're redirected to script. Just hit back button for now. We'll fix that up.

**Comment by crawleyde** [comment]
Great, thanks for this. We're setting up on our stage2.biblio site now.

**Comment by crawleyde** [comment]
@wendywarren 
--For this do I add Food Literacy as a parent term and Comfort Food as the child term?
Go to admin/structure/taxonomy/storybook and add Storybook Terms for Book and Chapter (including a la carte logo for Food Literacy (svg attached) and thumbnail (must be 142 x 80px) for Comfort Food.

**Comment by wendywarren** [comment]
@crawleyde  Yes, when you add Food Literacy (book) you will pick <root> for Relations, and for Comfort Food (chapter) you will pick Food Literacy for Relations. I hope that answered your question.

**Comment by crawleyde** [comment]
When I create terms on stage2, the Name field returns blank. This usually results from a mismatch between entity translation (which we have turned on for taxonomy terms) and the localize option. This is usually fixed by letting entity translation control term translation, which means setting the vocabulary to 'no multilingual options.' I'll try that.

**Comment by crawleyde** [comment]
Sorry, are "Book" and "Chapter" the two top-level terms? I think I missed that part.

**Comment by wendywarren** [comment]
Yep, the book will be the campaign (ex. Food Literact, Canada 150, etc..) that will be displayed across the top and the chapters (ex. Comfort Food) will be displayed in the dropdown.

![screen shot 2017-01-31 at 11 53 21 am](https://cloud.githubusercontent.com/assets/12865609/22475124/f7f823de-e7ab-11e6-8e0d-295fc340f6cf.png)

**Comment by crawleyde** [comment]
OK, so far so good. 
I reverted the taxonomy part of the feature, because changing the taxonomy to 'no multilingual options' deleted the taxonomy menu. I'd prefer to recreate the taxonomy menu if you can let me know briefly how to do that, or else if you could re-export with the taxonomy set to 'no multilingual.' (This seems to be the best option to get reliable translations, I've found.) If there's a reason not to do that, just let me know, thanks.
--I notice the width of the logo is a bit compressed, otherwise looking good so far.
![storybook](https://cloud.githubusercontent.com/assets/2026835/22477464/31a87c02-e7b4-11e6-9bfd-2e8d6a43e7f3.png)

**Comment by wendywarren** [comment]
@crawleyde - I've updated the storybook taxonomy to use 'No multilingual options for terms'. Going to commit that change now to the feature.

I also have the following checked at admin/config/regional/entity_translation - Taxonomy Term tab

![screen shot 2017-01-31 at 1 05 16 pm](https://cloud.githubusercontent.com/assets/12865609/22477979/1a2f333e-e7b6-11e6-9655-a10b5c5ba454.png)

This last change in configuration wasn't exported.

**Comment by crawleyde** [comment]
Great, thanks.

**Comment by wendywarren** [comment]
Morning @crawleyde! How are things going? Do you have any questions or concerns about the project?

**Comment by wendywarren** [comment]
@crawleyde I noticed the styling for the enlarged first letter is being applied to both paragraphs. I think I had always had just the one paragraph in that first Storybook article in the first region. I can adjust that in the css but you can also just break that second part into another article 'part'.

**Comment by wendywarren** [comment]
Here is the updated pdf guide. 

[oplstorybook.pdf](https://github.com/OPLBPO/opl_storybook/files/745342/oplstorybook.pdf)

**Comment by crawleyde** [comment]
Thanks @wendywarren -- we're putting test content into stage2. I think we're ok. I see a few template issues when I click on the logo on the test page: https://stage2.biblioottawalibrary.ca/en/storybook/food literacy/comfort food

Should I put this in a separate issue in the storybook module? 

```
    Notice: Undefined offset: 2 in opl_storybook_block_view() (line 174 of /home/stage2/sites/all/modules/custom/opl_storybook/opl_storybook.module).
    Notice: Trying to get property of non-object in opl_storybook_block_view() (line 174 of /home/stage2/sites/all/modules/custom/opl_storybook/opl_storybook.module).
    Notice: Undefined offset: 2 in opl_storybook_block_view() (line 175 of /home/stage2/sites/all/modules/custom/opl_storybook/opl_storybook.module).
    Notice: Trying to get property of non-object in opl_storybook_block_view() (line 175 of /home/stage2/sites/all/modules/custom/opl_storybook/opl_storybook.module).
    Notice: Undefined offset: 2 in opl_storybook_block_view() (line 176 of /home/stage2/sites/all/modules/custom/opl_storybook/opl_storybook.module).
    Notice: Trying to get property of non-object in opl_storybook_block_view() (line 176 of /home/stage2/sites/all/modules/custom/opl_storybook/opl_storybook.module).
    Notice: Undefined variable: base_url in include() (line 119 of /home/stage2/sites/all/themes/oplbootstrap/templates/page--storybook.tpl.php).
    Notice: Undefined property: view::$dom_id in template_preprocess_views_view() (line 116 of /home/stage2/sites/all/modules/contrib/views/theme/theme.inc).
```

**Comment by wendywarren** [comment]
@crawleyde Sure! I know what most of that is... from the next and previous chapter block. I'm not checking if there is a previous or next before I attempt to find values for them.

**Comment by wendywarren** [comment]
@crawleyde I noticed the audio content type isn't showing up again on the list to add content. That setting was changed here but didn't export: admin/structure/menu/manage/navigation

It's a matter of enabling the audio nav item.

I'm looking into why your audio player on stage2 looks different than the one I've been displaying.

**Comment by crawleyde** [comment]
FYI, I'm just in the process of setting up the storybook content on prod.

**Comment by crawleyde** [comment]
@minorOffense , @ClassicCut -- could I please get a second set of eyes on production -- enabling the storybook module package seems to be causing some 500 errors. I didn't see this on stage2 before. Thanks.

**Comment by crawleyde** [comment]
@coldfrontsupport A little help on prod, please?

**Comment by crawleyde** [comment]
Actually, I think the site's ok -- it went away for a while and was choking on all those module enables.

**Comment by wendywarren** [comment]
@crawleyde Is everything ok now Devin? I fixed up the theme for audio. Just about to commit my latest changes to oplbootstrap 7.x.

**Comment by crawleyde** [comment]
Yup, looks ok, thanks, just a big hiccup. I made a few small theme changes yesterday for the storybook webpage panel (not the storybook content itself). I'll do another prod update a bit later to get the latest changes.

**Comment by wendywarren** [comment]
Great!

**Comment by crawleyde** [comment]
Hi @wendywarren , I've updated stage2 with the latest merges, and I see the collection carousel seems to display all items: https://stage2.biblioottawalibrary.ca/en/storybook/food%20literacy/comfort%20food. I've cleared the cache a couple of times. This is in Firefox and Chrome on a Mac.

**Comment by crawleyde** [comment]
Also I think the js may be broken generally, since I don't see a fade in for the text.

**Comment by wendywarren** [comment]
@crawleyde Just checked it. JS is ok ... it's the max-width I put back in. I'll remove and commit. I'll msg you in a few minutes to pul the latest for theme.

**Comment by crawleyde** [comment]
Great, thanks.

**Comment by wendywarren** [comment]
@crawleyde Fix has been merged

**Comment by crawleyde** [comment]
Thanks, I'm just updating stage again.

**Comment by crawleyde** [comment]
Looks better, thanks. Would it be possible to change the URL to add dashes between the words so we can get rid of the empty spaces? Currently the URL looks a bit abnormal.

**Comment by wendywarren** [comment]
I asked Luc for some insight and he said that taxonomy terms don't have machine names by default. There may be a module that adds them. Otherwise, another option would to use taxonomy id for the url instead of taxonomy name.

**Comment by wendywarren** [comment]
I'll try taxonomy_machine_name and see how it goes

**Comment by crawleyde** [comment]
I was just wondering if there might be an option as in views to replace spaces with dashes in argument terms in the URL, and I was too lazy to check. I don't think it's worth adding another module for.

**Comment by wendywarren** [comment]
Panels is the one generating the url and argument settings only give term id or term name.

**Comment by crawleyde** [comment]
OK, let's just leave it then. Thanks @wendywarren

**Comment by crawleyde** [comment]
Interestingly, I see " Transform dashes in URL to spaces in term name filter values " is checked for both those name arguments, but isn't working when I add the dashes. I see some related issues in views. Not worth pursuing, but appears to be a Drupal bug that extends into 8.

**Comment by wendywarren** [comment]
I saw that too and tried it myself. I guess I had exported with it checked off. 

To fix the play controls showing in mobile browsers, I committed a quick fix this morning to oplbootstrap theme.
